packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -2896,3 +3025,270 @@ typedef SelectableRegionContextMenuBuilder = Widget Function(
+  /// {@macro flutter.widgets.ProxyWidget.child}
+  final Widget child;
+
+  @override
+  State<SelectionListener> createState() => _SelectionListenerState();
+}
+
+class _SelectionListenerState extends State<SelectionListener> {
+  late final _SelectionListenerDelegate _selectionDelegate = _SelectionListenerDelegate(selectionNotifier: widget.selectionNotifier);
+  _SelectableRegionSelectionStatusNotifier? _selectableRegionScope;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _selectableRegionScope?.removeListener(_handleOnSelectableRegionChanged);
+    _selectableRegionScope = _SelectableRegionScope.maybeOf(context);
+    _selectableRegionScope?.addListener(_handleOnSelectableRegionChanged);
+  }
+
+  @override
+  void didUpdateWidget(SelectionListener oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.selectionNotifier != widget.selectionNotifier) {
+      _selectionDelegate.setNotifier(widget.selectionNotifier);
+    }
+  }
+
+  @override
+  void dispose() {
+    _selectionDelegate.dispose();
+    super.dispose();
+  }
+
+  void _handleOnSelectableRegionChanged() {
+    if (_selectableRegionScope != null) {
+      _selectionDelegate._dispatchStatusEvent(_selectableRegionScope!.selectionListenerStatus);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SelectionContainer(
+      delegate: _selectionDelegate,
+      child: widget.child,
+    );
+  }
+}
+
+class _SelectionListenerDelegate extends _SelectableRegionContainerDelegate {
+  _SelectionListenerDelegate({
+    required SelectionListenerNotifier selectionNotifier,
+  }) : _selectionNotifier = selectionNotifier {
+    assert(
+      _selectionNotifier._selectionDelegate == null,
+      'The SelectionListenerNotifier provided is already attached to another [SelectionListener]. Try providing a new SelectionListenerNotifier.',
+    );
+    _selectionNotifier._selectionDelegate = this;
+  }
+
+  SelectionListenerNotifier _selectionNotifier;
+  void setNotifier(SelectionListenerNotifier newNotifier) {
+    _selectionNotifier = newNotifier;
+    _selectionNotifier._selectionDelegate = this;
+  }
+
+  SelectionGeometry? _lastFinalizedSelectionGeometry;
+  SelectionGeometry? _previousSelectionGeometry;
+
+  void _dispatchStatusEvent(SelectionListenerStatus status) {
+    if (_previousSelectionGeometry != value) {
+      _selectionNotifier.notifyListeners();
+      _selectionNotifier.notifyStatusListeners(status);
+    } else if (status == SelectionListenerStatus.finalized && _lastFinalizedSelectionGeometry != value) {
+      _selectionNotifier.notifyStatusListeners(status);
+      _lastFinalizedSelectionGeometry = value;
+    }
+    _previousSelectionGeometry = value;
+  }
+
+  @override
+  void dispose() {
+    _selectionNotifier._unregisterSelectionListenerDelegate();
+    super.dispose();
+  }
+}
+
+/// Notifies listeners when the selection under a [SelectionListener] has been
+/// changed or finalized.
```

Comment:
Nit: Maybe mention that this is typically passed into [SelectionListener.selectionListener]? If that's not overly obvious. I'm thinking about this because that's pretty much the only way to actually use SelectionListenerNotifier right now, right?

Reply:
Good point, yeah SelectionListenerNotifier can only be passed and used by SelectionListener right now.